exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -43,6 +45,42 @@ import (
 
 var jsonListPrefix = []byte("[")
 
+// objsFromYamlData parses a collection of yaml documents into Unstructured objects.
+// The returned objects are sorted for creation priority within the objects defined
+// in the same document. The flattening of the documents preserves the original order.
+func objsFromYamlData(yamlDocs [][]byte) ([]unstructured.Unstructured, error) {
+	allObjs := []unstructured.Unstructured{}
+	for _, data := range yamlDocs {
+		isJSONList, err := isJSONList(data)
+		if err != nil {
+			return nil, err
+		}
+		objs := []unstructured.Unstructured{}
+		// If it is a json list, convert each list element to an unstructured object.
+		if isJSONList {
+			var results []map[string]interface{}
+			// Unmarshal the JSON to the interface.
+			if err = json.Unmarshal(data, &results); err == nil {
+				for i := range results {
+					var u unstructured.Unstructured
+					u.SetUnstructuredContent(results[i])
+					objs = append(objs, u)
+				}
+			}
+		} else {
+			// If it is not a json list, data is either json or yaml format.
+			objs, err = utilyaml.ToUnstructured(data)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed converting data to unstructured objects")
+			}
+		}
+
+		allObjs = append(allObjs, utilresource.SortForCreate(objs)...)
```

Comment:
Good question. I'm not opposed but that would be a change in behavior.

I think the idea was that all entries in the data field of a `ConfigMap` or `Secret` are treated as separate documents. The object of each document are sorted for create, but only in relation with the objects of that document. Then the different documents are sorted by their keys in the map (I presume to achieve repeatable behavior).

I'm not completely opposed to change it, but since I don't really see an advantage in sorting all objects across documents, I would vote to keep it as is, so we maintain the previous behavior and avoid introducing errors for users that already have CRS's.

Reply:
I think I'm ok with keeping the current ordering per yamlDoc, I'm not aware of this causing problems to users and if possible I would prefer to keep this PR scoped to the new reconcile mode change

Reply:
My bad. I think I brought this up because I thought this behavior changed with this PR, but it didn't